### DISCUSSION TITLE
Allowing the use of .js / .ts inputs (over json)

### DISCUSSION THREAD

**Discussion by iamnbutler** [six7/figma-tokens#1057] [Ideas]
Being able to use `.js/.ts` for token files would enable a bunch of useful things like documenting tokens using doc comments, interfacing easier with javascript projects, etc. I couldn't see any discussions here or in the issues around this, but I imagine I'm not the only one that would want this.

At [Zed](https://zed.dev/) we use custom typescript app to theme the Zed UI, it would be really nice not to have to generate separate tokens for use with Figma. Being able to add doc comments means it would be easier to dynamically generate documentation for tokens as well.

Curious if anyone else is thinking about this.

---
For search: input, javascript, typescript, comments

**Comment by six7** [comment]
One issue I'd see is that we'd need to know what type of file users are using when they have multi-file sync enabled. Might be needed anyway as in the future the W3C spec foresees both `.json` and `.tokens` or even `.tokens.json` as being a filetype. I could see us using a regex of file types to scan for.

I'd assume inside the file tokens would also be stored differently?

What's the big advantage over having a style dictionary transformation in place that takes the tokens.json and creates a .ts file to be consumed by typescript? A token's `description` field could be transformed to the doc comments that you mention, while staying in the standardized format.
